DependencyInjection/Compiler/StrictPass.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\IntlBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Reference;
+
+class StrictPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $def = $container->getDefinition('sonata.intl.locale_detector.request');
+        $def->replaceArgument(0, $this->changeReference($def->getArgument(0), 'service_container'));
+
+        $def = $container->getDefinition('sonata.intl.locale_detector.request');
+        $def->replaceArgument(0, $this->changeReference($def->getArgument(0), 'session'));
+    }
+
+    /**
+     * @param Reference $reference
+     * @param string    $serviceId
+     *
+     * @return Reference
+     */
+    private function changeReference(Reference $reference, $serviceId)
+    {
+        // Stay compatible with Symfony 2
+        if (method_exists($reference, 'isStrict')) {
```

Comment:
Note that `isStrict(false)` won't trigger a deprecation warning in Symfony 2.8: https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/DependencyInjection/Reference.php#L66

And earlier Symfony versions do not have the deprecation warning.